var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -117,6 +120,9 @@ def cmake_args(self):
             options.append('-DUSolids_DIR=%s' % spec[
                 'vecgeom'].prefix.lib.CMake.USolids)
 
+        on_or_off = lambda opt: 'ON' if '+' + opt in spec else 'OFF'
```

Comment:
Apparently `CMakePackage` could use functionality analogous to `AutotoolsPackage.with_or_without`, this looks to be the way to do it for now.

Reply:
Yeah, I [mentioned this](
https://github.com/spack/spack/pull/9409#discussion_r222456324) as well -- I'm not sure what to call it (`on_or_off`, `with_or_without`, `enable_or_disable`) and it will have to work a little differently than the autotools version. If I have some time I'll choose one and make a PR in which we can discuss it 😄